core/include/scipp/core/multi_index.h

```diff
@@ -189,11 +206,15 @@ template <scipp::index N> class MultiIndex {
 
   [[nodiscard]] constexpr scipp::index
   inner_distance_to(const MultiIndex &other) const noexcept {
+    // The end index has coord[0] = 0.
+    if (other.m_coord[0] == 0) {
+      return inner_distance_to_end();
+    }
     return other.m_coord[0] - m_coord[0];
```

Comment:
Can you explain the reason behind this change?

Reply:
Suppose we have a shape of `[4, 3, 2]` and no chunking in transform. A possible begin index index for the inner loop has `coord = [0, 1, 0]` and the corresponding end index has `coord = [0, 2, 0]`. The 'inner distance' between the two is `4` because they cover the whole inner dimension. But the check in line 213 would return 0 which produces an infinite loop.

This is the bug that only appears in a serial build because with chunking, the end index normally does not have `coord[0] = 0`.

Reply:
But shouldn't we get `indices.in_same_chunk(end, 1) == false` then and not call this function?

Reply:
Yes it should and now it does. There was a `+1` missing in `in_same_chunk`. See the latest changes